src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size++] = value;
+    }
 
+    private void resize() {
+        int newCapacity = (int) (elements.length * 1.5);
```

Comment:
Any magic numbers should be constants. The growth factor '1.5' should be defined as a constant with a descriptive name.